projects/hathor-wallet-mobile/0002-nano-contract-integration.md

```diff
@@ -0,0 +1,591 @@
+
+```json
+{
+  "id": 3,
+  "jsonrpc": "2.0",
+  "method": "htr_createNcTx",
+  "params": {
+	"bluprintId": "494d0ac59a6918b771b122395206fef3f349f84f20dc430188a319d4ead24a3b",
+	"method": "swap",
+	"caller": "WRqBSi5jc74P8tRo7CmgiGByA1eTzfMh7B",
+	"ncId": "000001e43dda7348c53c6252d98799c9e2a0b4ea3b5f10b1182578d643c9641d",
+	"ncActions": [
+		{
+			"type": "deposit",
+			"token": "000001a176aa72b84f3486b3fcea327b023a70d1b6461c1b6a9073785b5775d8",
+			"data": {
+				"amount": "100",
+			},
+		},
+		{
+			"type": "withdrawal",
+			"token": "000001a176aa72b84f3486b3fcea327b023a70d1b6461c1b6a9073785b5775d8",
+			"data": {
+				"amount": "200",
+				"address": "WRqBSi5jc74P8tRo7CmgiGByA1eTzfMh7B",
+			},
+		},
+	],
+	"ncArgs": [],
+}
+```
+
+### Nano Contract Transaction Request
+
+##### When execute
+- Every time the a connected dapp send the request `htr_createNcTx`.
+
+##### Where to execute
+- It must be handled on `sagas/walletConnect.js@onSessionRequest` and dispatched to `onCreateNcTxRequest`.
+- It must be processed on `sagas/walletConnect.js@onCreateNcTxRequest`.
+
+##### How to execute `onCreateNcTxRequest`
+1. Validate the transaction request
+2. Ask user for `Approval` or `Rejection`
+3. If rejected, return to dapp an error message
+4. If approved, ask user's Pin Code
+5. Create and send transaction
+6. Return to dapp a success message 
+
+##### What render
+- On user's `Approval` or `Rejection` a modal must shows up over any screen to presents all the request details specified on [`htr_createNcTx`](#htr_createnctx) driven by design of [Modal: Nano Contract Transaction Request](modal-nano-contract-transaction-request).
+- The modal component must be registered on `src/components/WalletConnect`.
+
+# Draw backs
+
+## Nano Contract Balance
+
+We shouldn't load Nano Contract Balance because Nano Contracts can be much different from each other and the meaning of balance depends on the Nano Contract purpose. For instance, what means a balance on a *Bet* Nano Contract? Nothing, because each bet is tyied up to a single address; and what say about Swap? Also nothing, because it doens't hold user's assets.
+
+# Rationale and alternatives
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+## [Alternative] Component: Nano Contract Transaction List
+[alternative-component-nano-contract-transaction-list]: #alternative-component-nano-contract-transaction-list
+
+This alternative displays a list of transaction items in which each item represents the transaction as whole, not only part of it segregated by type. By clicking on the item the user is presented with all the transaction details, including each token balance involved.
+
+## Feature onboarding
+[feature-onboarding]: #feature-onboarding
+
+Presents an onboarding with 3 or 4 views explaining the Nano Contract feature, its functionalities and how to use. The onboard can appears after the toggle button activation and in the end ask the user to proceed with wallet reloading.
```

Comment:
Well, I don't see any problem to have both, but the nature of either solutions are very different. Thinking only in the user's experience during the wallet usage, I prefer to keep continuity of the experience, therefore the onboarding better suits this principal. I believe it would be more fun and less expensive in terms of attention to users.

Reply:
@anafigueired Can you give your view on this matter? You were the one that proposed it.